src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2239,3 +2239,41 @@ Base.getindex(::AbstractDataFrame, ::Union{Symbol, Integer, AbstractString}) =
 
 Base.setindex!(::AbstractDataFrame, ::Any, ::Union{Symbol, Integer, AbstractString}) =
     throw(ArgumentError("syntax df[column] is not supported use df[!, column] instead"))
+
+"""
+    reverse(df::AbstractDataFrame)
+
+This function reverses the current order of the rows in `df`.
```

Comment:
This function should create a new data frame with reversed rows. `reverse!` would reverse rows in the passed data frame in-place.